Lib/typing.py

```diff
@@ -297,8 +305,27 @@ def _check_generic(cls, parameters, elen):
         raise TypeError(f"{cls} is not a generic class")
     alen = len(parameters)
     if alen != elen:
-        raise TypeError(f"Too {'many' if alen > elen else 'few'} arguments for {cls};"
-                        f" actual {alen}, expected {elen}")
+        expect_val = elen
+        if hasattr(cls, "__parameters__"):
+            parameters = [p for p in cls.__parameters__ if get_origin(p) is not Unpack]
```

Comment:
It's not immediately clear to me what the implications are of excluding `Unpack` here, or why it's correct to do so. Are there tests related to this? (I don't see `Unpack` usage added to `test_typing.py`.) It seems like this is worth tests and a comment.

It also seems a bit strange to me in terms of division of responsibilities and maintainability that `_check_generic`, which is called in various cases, would have special-case code in it that appears to be only relevant to the callsite in `_generic_class_getitem`, and uses `hasattr` to detect whether it's relevant. Would it be reasonable to extract this code to a separate function and extend the signature of `_check_generic` to e.g. take both a minimum and maximum number of expected parameters? The minimum could even be passed as an optional callback, if there's a concern about inefficiency of unnecessarily checking defaults when all parameters have been provided.

Reply:
Thanks, I realized the implementation for TypeVarTuple defaults is wrong in various places; I need to make a number of changes.

Reply:
I pushed some changes to make TypeVarTuple work; I need to add more test cases though. I also posted on https://discuss.python.org/t/pep-696-type-defaults-for-typevarlikes/22569/36 to get more opinions on what the grammar should look like.

(I wasn't too keen on adding support for TypeVarTuple defaults in the first place. I doubt that part of the feature is going to get much use.)